video/transmux.go

```diff
@@ -95,21 +97,89 @@ func MuxTStoFMP4(fmp4ManifestOutputFile string, inputs ...string) error {
 	return nil
 }
 
-func ConcatTS(tsFileName string, segmentsList *TSegmentList) (int64, error) {
+func ConcatTS(tsFileName string, segmentsList *TSegmentList, useStreamBasedConcat bool) (int64, error) {
 	var totalBytes int64
-	// 1. create a .ts file for a given rendition
-	tsFile, err := os.Create(tsFileName)
-	if err != nil {
-		return totalBytes, fmt.Errorf("error creating file (%s) err: %w", tsFileName, err)
-	}
-	defer tsFile.Close()
-	// 2. for a given rendition, write all segment indices in ascending order to the single .ts file
-	for _, k := range segmentsList.GetSortedSegments() {
-		segBytes, err := tsFile.Write(segmentsList.SegmentDataTable[k])
+	if !useStreamBasedConcat {
```

Comment:
This block is becoming pretty huge, could we split each part out into methods?

Reply:
Sure, I split out the ffmpeg bits into it's own separate fcn and left file management bits in the main fcn.